lesson18/index.js

```diff
@@ -0,0 +1,186 @@
+
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+    }
+    return json.data;
+}
+
+async function fetchImgData() {
+    return new Promise((resolve) => {
+        setTimeout(() => resolve(fetchData("https://mocki.io/v1/2db56900-1359-4d61-bd4c-d477c01b0122")), 3000);
+    });
+}
+
+async function getImgData() {
+    addLoading();
+    try {
+        return await fetchImgData();
+    } catch(e) {
+        createErrorMessage(e);
+    } finally {
+        removeLoading();
+    }
+}
+
+function createErrorMessage(error) {
+    const p = document.createElement("p");
+    p.classList.add("slideshow__error");
+    p.textContent = error;
+    console.error(error);
+    ul.appendChild(p);
+}
+
+function renderListOfImg(data) {
+    const fragment = document.createDocumentFragment();
+
+    for (let i = 0; i < data.length; i++) {
+        const li = document.createElement("li");
+        const img = document.createElement("img");
+
+        li.classList.add("slideshow__pict-item","js-slideshow-item");
+        li.dataset.index = `${i}`;
+        img.src = data[i].img;
+        img.alt = data[i].alt;
+    
+        //JSONデータでdisplay:trueの場合はis-activeを付与
+        data[i].display && li.classList.add("is-active");
+
+        fragment.appendChild(li).appendChild(img);
+    }
+    ul.appendChild(fragment);
+}
+
+function renderListOfPagination(data) {
+    const paginationList = document.getElementById("js-pagination-list");
+    const fragment = document.createDocumentFragment();
+
+    for (let i = 0; i < data.length; i++) {
+        const li = document.createElement("li");
+        const Button = document.createElement("button");
+
+        li.classList.add("pagination__item","js-pagination-item");
+        li.dataset.index = `${i}`;
+        Button.classList.add("pagination__btn","js-pagination-btn");
+        Button.dataset.index = `${i}`;
+
+        //JSONデータでdisplay:trueの場合はis-activeを付与
+        data[i].display && li.classList.add("is-active");
+
+        fragment.appendChild(li).appendChild(Button);
+    }
+    pagination.insertAdjacentElement("afterbegin",paginationList).appendChild(fragment);
+}
+
+function init(data) {
+    const allofCountElement = document.getElementById("js-counter-all");
+    
+    allofCountElement.textContent = data.length;
+    incrementCurrentIndex(getCurrentIndex());
+    toggleButtonDisabled(data.length);
+}
+
+function toggleButtonDisabled(length) {
+    const currentIndex = getCurrentIndex();
+    const firstIndex = 0;
+    const lastIndex = length - 1;
+
+    previousButton.disabled = currentIndex === firstIndex;
+    nextButton.disabled = currentIndex === lastIndex;
+}
+
+function getCurrentIndex() {
+    const activeImg = ul.querySelector(".is-active");
+    return Number(activeImg.dataset.index);
+}
+
+function incrementCurrentIndex(number) {
+    const currentCountElement = document.getElementById("js-counter-current");
+    currentCountElement.textContent = number + 1;
```

Comment:
IMO
If you name a function increment, it would be better if you used the increment operator, what do you think?
[インクリメント](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Increment)
```suggestion
    currentCountElement.textContent = ++number;
```

Reply:
Thanks for telling us about the increment! 
I learned that the meaning of the prefix and the postfix are different,
and I am very glad to know that!
I fixed it.
[c729b67](https://github.com/haru-programming/JavaScript-lessons/pull/22/commits/c729b674423d12308431356dc609d8107b023b2c)